test/integration/render-tests/line-gradient/gradient-step-zoomed-cross-continental/style.json

```diff
@@ -0,0 +1,98 @@
+{
+  "version": 8,
+  "metadata": {
+    "test": {
+      "width": 256,
+      "height": 64,
+      "allowed": 0.0015
+    }
+  },
+  "zoom": 22,
+  "center": [
+    -77.0336537,
+    38.89992645
+  ],
+  "sources": {
+    "gradient": {
+      "type": "geojson",
+      "data": {
+        "type": "FeatureCollection",
+        "features": [
+          {
+            "type": "Feature",
+            "geometry": {
+              "type": "LineString",
+              "coordinates": [
+                [
+                  -77.033643,
+                  38.899926
+                ],
+                [
+                  -120.033643,
+                  38.899926
+                ]
```

Comment:
This straight line is 3&nbsp;686&nbsp;kilometers long. Assuming the intention is to support gradient lines of up to the same distance in gl-native for use in the Mapbox Navigation SDK, it might be worth considering that the Mapbox Directions API can return a route [up to 10&nbsp;000&nbsp;kilometers long](https://docs.mapbox.com/api/navigation/#directions-api-restrictions-and-limits). I’m not sure how common that would be in practice, however. Another mitigating factor would be that the navigation SDK cuts the route line up by leg.

Reply:
Yes this test is for testing potential visual regression, not necessarily stress testing the issues we're trying to overcome. I wouldn't see an issue with having routes that long.